File: sorting/bubbleSort.py
class Bubblesort:
    
   def bubble (self,arr):
      
      self.n=len(arr)
      flag=False  # to check swapping is necessary or not or already list is swaped 
      self.count=0
    
      for i  in range (0,self.n):
        for j in range (0,self.n-1-i):
            if (arr[j]>arr[j+1]):
                flag=True
                temp=arr[j]
                arr[j]=arr[j+1]
                arr[j+1]=temp
                self.count+=1

        if (flag==False):  # if the swaaping not happend it means loop is already sorted 
           break     # break statement break the innner  and outer both the  looop 

        
    

arr=[5,4,3,2,1]

print(arr)
sort=Bubblesort()
sort.bubble(arr)




print(arr)
print(sort.count)  # object k through acces kar sakta h fn ko 4

File: sorting/bubbleSort.java
package repss.sorting;
import java.util.*;


public class bubbleSort{


    public static void bubble(int[] arr){
        int n= arr.length;

        for (int i=0; i<n-1;i++){
            for (int j=0; j<n-i-1;j++){
                
                if (arr[j]> arr[j+1]){
                    int temp= arr[j];
                    arr[j]=arr[j+1];
                    arr[j+1]=temp;

                }
            }
        }
    }

    public static void print(int arr[]){
        for (int i=0; i<arr.length;i++){
            System.out.print(arr[i]+" ");
        }
    }
    public  static void main (String [] args){

    
        System.out.print("****Bubble sort logic \n logic start  from 0 and send the element  and if element is bigeer then swap the position ");

        int arr[]={5,4,3,1,2};


        print(arr);
        System.out.println();
        bubble(arr);

        print(arr);





    }

}
